src/pybind/pyembed.cpp

```diff
@@ -0,0 +1,85 @@
+/*************************************************************************
+ * Copyright (C) 2018-2019 Blue Brain Project
+ *
+ * This file is part of NMODL distributed under the terms of the GNU
+ * Lesser General Public License. See top-level LICENSE file for details.
+ *************************************************************************/
+
+#include <cstdlib>
+#include <dlfcn.h>
+
+#include "pybind/pyembed.hpp"
+#include "utils/logger.hpp"
+
+namespace nmodl {
+
+namespace pybind_wrappers {
+
+bool EmbeddedPythonLoader::have_wrappers() {
+#if defined(STATIC_PYWRAPPER)
+    static auto wrapper_api = nmodl::pybind_wrappers::init_pybind_wrap_api();
+    wrappers = &wrapper_api;
+    return true;
+#else
+    wrappers = static_cast<pybind_wrap_api*>(dlsym(RTLD_DEFAULT, "nmodl_wrapper_api"));
+    return wrappers != nullptr;
+#endif
+}
+
+void EmbeddedPythonLoader::load_libraries() {
+    const auto pylib_env = std::getenv("NMODL_PYLIB");
+    if (!pylib_env) {
+        logger->critical("NMODL_PYLIB environment variable must be set to load embedded python");
+        throw std::runtime_error("NMODL_PYLIB not set");
+    }
```

Comment:
yesterday when I was building this, I was seeing exception at the time of `make` and build was finishing without error status and then there were 26 tests instead of 27.
Is it the same behaviour? If so, do you think something easily can be done to propagate the error?

Reply:
That's exactly the issue that I had explained earlier in our slack conversation. What happens is that during build, when a test executable is built, catch2 will then run it with `--list-tests`, if the execution fails for some reason that error is not propagated from the CatchAddTests cmake_command further up to `make` and the build does not fail. I believe this is a bug in `CatchAddTests.cmake`. Shall I try and fix it locally?
What is happening in your case is that the python wrapper is not found, so we proceed with trying to load lipython. If at that point we don't find the environment variable or libpython, then we throw an exception and the process terminates. So that behavior is correct, I think.

Reply:
>  I believe this is a bug in CatchAddTests.cmake. Shall I try and fix it locally?

I assume we will use this LINK_AGAINST_PYTHON=OFF mode while only building wheels, so I think this is not on critical path and blocker for this PR. (this improvement we can do later, we can just create an issue)